src/diffusers/models/transformer_2d.py

```diff
@@ -235,7 +235,7 @@ def __init__(
 
         self.caption_projection = None
         if caption_channels is not None:
-            self.caption_projection = CaptionProjection(in_features=caption_channels, hidden_size=inner_dim)
+            self.caption_projection = PixArtAlphaTextProjection(in_features=caption_channels, hidden_size=inner_dim)
```

Comment:
Might actually be worth breaking up Transformer2D up into a dedicated one for PixArt.

Reply:
For a future PR, yeah? I am happy to work on it once this is merged.

Reply:
Definitely for a future PR

But I think we should refactor transformers and UNet after we clean up all the lower-level classes and make such decisions all together so it will be consistent

Reply:
Yes 100 percent!